packages/openapi-generator/src/sdk-metadata/sdk-metadata.ts

```diff
@@ -0,0 +1,25 @@
+import {
+  getPregeneratedLibrary,
+  getSdkMetadataClient
+} from '@sap-cloud-sdk/generator-common';
+import type { Client } from '@sap-cloud-sdk/generator-common';
+import { OpenApiDocument } from '../openapi-types';
+import { GeneratorOptions } from '../options';
+import { packageDescription } from './package-description';
+import { getGenerationAndUsage } from './generation-and-usage';
+
+export async function sdkMetaDataJS(
+  openApiDocument: OpenApiDocument,
+  options: GeneratorOptions
+): Promise<Client> {
+  const [pregeneratedLibrary, generationAndUsage] = await Promise.all([
+    getPregeneratedLibrary(
+      packageDescription(openApiDocument.npmPackageName),
```

Comment:
I think we should add a follow up ticket to simply the description. For OData we use in one if the s4HanaCloud option is there some specific test. However, if we start creating and publishing also other OData services we need to adjust this anyway.